usaspending_api/etl/management/commands/load_table_from_delta.py

```diff
@@ -0,0 +1,288 @@
+            if postgres_model:
+                postgres_cols = [(column.name, type(column)) for column in postgres_model._meta.get_fields()]
+            else:
+                postgres_cols = list(postgres_cols.items())
+            for column_name, column_type in postgres_cols:
+                if column_type in SPECIAL_TYPES_MAPPING:
+                    special_columns[column_name] = column_type
+
+        # Read from Delta
+        df = spark.table(delta_table)
+
+        # Make sure that the column order defined in the Delta table schema matches
+        # that of the Spark dataframe used to pull from the Postgres table. While not
+        # always needed, this should help to prevent any future mismatch between the two.
+        if table_spec.get("column_names"):
+            df = df.select(table_spec.get("column_names"))
+
+        # If we're working off an existing table, truncate before loading in all the data
+        if not make_new_table:
+            with db.connection.cursor() as cursor:
+                cursor.execute(f"TRUNCATE {temp_table}")
+
+        # Write to Postgres
+        logger.info(f"LOAD (START): Loading data from Delta table {delta_table} to {temp_table}")
+        split_dfs = self._split_dfs(df, list(special_columns))
+        split_df_count = len(split_dfs)
+        for i, split_df in enumerate(split_dfs):
+            # Note: we're only appending here as we don't want to re-truncate or overwrite with multiple dataframes
+            logger.info(f"LOAD: Loading part {i+1} of {split_df_count} (note: unequal part sizes)")
+            split_df.write.jdbc(
+                url=jdbc_url,
+                table=temp_table,
+                mode="append",
+                properties=get_jdbc_connection_properties(),
+            )
+            logger.info(f"LOAD: Part {i+1} of {split_df_count} loaded (note: unequal part sizes)")
+        logger.info(f"LOAD (FINISH): Loaded data from Delta table {delta_table} to {temp_table}")
+
+        # We're done with spark at this point
+        if spark_created_by_command:
+            spark.stop()
+
+        # Load indexes and constraints if applicable
+        if postgres_table and (copy_indexes or copy_constraints):
+            copy_table_metadata_args = [
+                "--source-table",
+                postgres_table,
+                "--dest-table",
+                temp_table,
+            ]
+            if not copy_constraints:
+                copy_table_metadata_args.append("--skip-constraints")
+            if not copy_indexes:
+                copy_table_metadata_args.append("--skip-indexes")
+            for option in ["max_parallel_maintenance_workers", "maintenance_work_mem", "index_concurrency"]:
+                if options[option]:
+                    copy_table_metadata_args.extend([f'--{option.replace("_", "-")}', options[option]])
+            call_command("copy_table_metadata", *copy_table_metadata_args)
```

Comment:
Aren't we trying to avoid this because it will be using cluster resources for database intensive tasks? Any reason to keep this logic here if we're not planning on using it in the pipeline?

Reply:
Generally yes. I kept it in as a kind of reminder that it's necessary that these should be done in succession and an option if we didn't run this in a pipeline. By default it won't but it could be used if someone wanted to kick it off and not have to come back to run the next command. I can take it out and leave a note if we don't think we'll ever need that option.

Reply:
Took it out and included an additional log.

Reply:
Logging looks good. Nice touch to add the necessary parameters for the next step